raiden_contracts/deploy/contract_verifier.py

```diff
@@ -273,19 +273,25 @@ def _verify_user_deposit_deployment(
     monitoring_service_address: Address,
 ):
     """ Check an onchain deployment of UserDeposit and constructor arguments at deployment time """
-    assert len(constructor_arguments) == 2
-    assert to_checksum_address(user_deposit.functions.token().call()) == token_address
-    assert token_address == constructor_arguments[0]
-    assert user_deposit.functions.whole_balance_limit().call() == user_deposit_whole_balance_limit
+    if len(constructor_arguments) != 2:
+        raise RuntimeError("UserDeposit has a wrong number of constructor arguments.")
+    if token_address != constructor_arguments[0]:
+        raise RuntimeError("UserDeposit received a wrong token address during construction.")
+    if to_checksum_address(user_deposit.functions.token().call()) != token_address:
+        raise RuntimeError("UserDeposit has a wrong token address onchain.")
+    if user_deposit.functions.whole_balance_limit().call() != user_deposit_whole_balance_limit:
+        raise RuntimeError("UserDeposit has a wrong whole_balance_limit onchain")
+    if user_deposit_whole_balance_limit != constructor_arguments[1]:
+        raise RuntimeError("UserDeposit received a wrong whole_balance_limit during construction.")
     assert user_deposit_whole_balance_limit == constructor_arguments[1]
-    assert (
-        to_checksum_address(user_deposit.functions.one_to_n_address().call()) == one_to_n_address
-    )
+    if to_checksum_address(user_deposit.functions.one_to_n_address().call()) != one_to_n_address:
```

Comment:
You could also use `eth_utils.is_same_address` instead of converting them. But the advantage of your way is that you make sure it's actually a checksum address.

Reply:
I didn't think about that.  Maybe `is_same_address` says more precisely what I wanted to check here.